gems/pending/OpenStackExtract/MiqOpenStackVm/MiqOpenStackInstance.rb

```diff
@@ -30,7 +30,7 @@ def instance
   end
 
   def snapshot_metadata
-    @snapshot_metadata ||= instance.metadata.get(:miq_snapshot)
+    @snapshot_metadata ||= instance.metadata.length > 0 && instance.metadata.get(:miq_snapshot)
```

Comment:
The fact that `instance.metadata.length > 0` doesn't ensure that the `:miq_snapshot` metadata item exists.
It's probably better to `rescue` the specific not found error and set `@snapshot_metadata` to `nil`.